hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/metrics2/sink/PrometheusMetricsSink.java

```diff
@@ -98,22 +108,25 @@ public void writeMetrics(Writer writer) throws IOException {
     for (Map.Entry<String, Map<Collection<MetricsTag>, AbstractMetric>> promMetric :
         promMetrics.entrySet()) {
       AbstractMetric firstMetric = promMetric.getValue().values().iterator().next();
+      List<String> customTags = new ArrayList<>();
```

Comment:
The ArrayList instance is created for each metric. Would you create an instance outside of the for-loop and reuse it?

Reply:
done. change customTags to extendMetricsTags to make it more readable and reuse it.